Sources/StreamChat/Repositories/OfflineRequestsRepository.swift

```diff
@@ -40,20 +40,21 @@ class OfflineRequestsRepository {
     /// - If the request fails with a connection error -> The request is kept to be executed once the connection is back (we are not putting it back at the queue to make sure we respect the order)
     /// - If the request fails with any other error -> We are dismissing the request, and removing it from the queue
     func runQueuedRequests(completion: @escaping () -> Void) {
-        var pendingActions: [(String, Data)] = []
-
         let readContext = database.backgroundReadOnlyContext
-        readContext.performAndWait {
-            pendingActions = QueuedRequestDTO.loadAllPendingRequests(context: readContext).map {
-                ($0.id, $0.endpoint)
+        readContext.perform { [weak self] in
+            let requests = QueuedRequestDTO.loadAllPendingRequests(context: readContext).map { ($0.id, $0.endpoint) }
+            DispatchQueue.main.async {
```

Comment:
Why do we dispatch in the main queue here?

Reply:
Because we don't want to continue the execution in the NSManagedObjectContext's thread

Reply:
But the `executeRequests` has a DispatchGroup which waits for all the requests. Which means we will block the main thread until all requests are made. Or Am I missing something?

Reply:
I see... the requests are then executed in the different operation queue, but then dispatching it to the main queue also does not seem to do anything. Was there a problem before? I was this done?

Reply:
The change here was only to avoid the performAndWait, which could be error prone, and I did not want to risk it as I saw a crash that could look like a deadlock. Because of its async nature, now I cannot wait on the method's callee thread, so to be safe I just resume on the main thread